Simplenote/PublishViewController.swift

```diff
@@ -0,0 +1,169 @@
+import Foundation
+
+
+// MARK: - Represents the PublishViewController's Internal State
+//
+enum PublishState {
+    case publishing
+    case published(url: String)
+    case unpublishing
+    case unpublished
+}
+
+// MARK: - VersionsViewControllerDelegate
+//
+@objc
+protocol PublishViewControllerDelegate {
+    func publishControllerDidClickPublish(_ controller: PublishViewController)
+}
+
+
+// MARK: - PublishViewController
+//
+class PublishViewController: NSViewController {
+
+    /// Publish Legend
+    ///
+    @IBOutlet private var legendTextField: NSTextField!
+
+    /// Publish Button
+    ///
+    @IBOutlet private var publishButton: NSButton!
+
+    /// Note's Published URL
+    ///
+    @IBOutlet private var urlTextField: NSTextField!
+
+    /// Internal State
+    ///
+    private var state: PublishState = .unpublishing {
+        didSet {
+            refreshInterface(newState: state)
+        }
+    }
+
+    /// Returns the Publish Button's Internal State
+    ///
+    @objc
+    var publishButtonState: NSControl.StateValue {
+        publishButton.state
+    }
+
+    /// Old School delegate
+    ///
+    weak var delegate: PublishViewControllerDelegate?
```

Comment:
This guy is probably gonna want a `@objc` so `NoteEditorViewController` can see it ;)